File: KempIS Backend/src/Application/Reception/Realtime/ReceptionEventNames.cs
namespace Application.Reception.Realtime;

public static class ReceptionEventNames
{
  public const string PairJoin = "pair:join";
  public const string PairReady = "pair:ready";
  public const string PairPeerLeft = "pair:peer_left";
  public const string PairDisplaced = "pair:displaced";
  public const string Error = "error";

  public const string SessionPush = "session:push";
  public const string SessionClear = "session:clear";
  public const string SignatureCaptured = "signature:captured";
  public const string SignatureCleared = "signature:cleared";
  public const string GuestSigned = "guest:signed";
  public const string GuestSignatureCleared = "guest:signature_cleared";
  public const string EDokladyStart = "edoklady:start";
  public const string EDokladyTransaction = "edoklady:transaction";
  public const string EDokladyState = "edoklady:state";
  public const string EDokladyResult = "edoklady:result";
  public const string EDokladyCancel = "edoklady:cancel";
}

File: KempIS Frontend/src/app/core/reception-realtime/reception-event-names.ts
/** Ported verbatim from
 *  `Application.Reception.Realtime.ReceptionEventNames`. Any divergence
 *  is a bug. */
export const ReceptionEventNames = {
  // Sent by both peers; inspected by server, NOT relayed.
  PairJoin: "pair:join",

  // Server-emitted (not relayed; server is source).
  PairReady: "pair:ready",
  PairPeerLeft: "pair:peer_left",
  PairDisplaced: "pair:displaced",
  Error: "error",

  // Desktop → tablet (relayed).
  SessionPush: "session:push",
  SessionClear: "session:clear",
  GuestSigned: "guest:signed",
  GuestSignatureCleared: "guest:signature_cleared",
  EdokladyTransaction: "edoklady:transaction",
  EdokladyState: "edoklady:state",
  EdokladyResult: "edoklady:result",

  // Tablet → desktop (relayed).
  SignatureCaptured: "signature:captured",
  SignatureCleared: "signature:cleared",
  EdokladyStart: "edoklady:start",

  // Sendable by either peer (relayed).
  EdokladyCancel: "edoklady:cancel",
} as const;

export type ReceptionEventName =
  (typeof ReceptionEventNames)[keyof typeof ReceptionEventNames];

export type EdokladyState =
  | "Open"
  | "WaitingForResponse"
  | "ResponseReceived"
  | "Finished"
  | "Failed"
  | "Canceled"
  | "Unfinished"
  | "Timeout";

export type EdokladyOutcome =
  | "Success"
  | "Untrusted"
  | "UnknownError"
  | "MissingData"
  | "Expired";

export type ReceptionErrorCode =
  | "invalid_pair_code"
  | "role_taken"
  | "not_paired"
  | "payload_too_large"
  | "bad_request";

export type PeerRole = "desktop" | "tablet";
